diff --git a/sdk/python/feast/grpc/grpc.py b/sdk/python/feast/grpc/grpc.py
index 28b21437..7ba8b8de 100644
--- a/sdk/python/feast/grpc/grpc.py
+++ b/sdk/python/feast/grpc/grpc.py
@@ -41,9 +41,19 @@ def create_grpc_channel(
             credentials = grpc.composite_channel_credentials(
                 credentials, grpc.metadata_call_credentials(auth_metadata_plugin),
             )
-        channel = grpc.secure_channel(url, credentials=credentials)
+        channel = grpc.secure_channel(url, credentials=credentials,
+            options=[
+                ('grpc.max_send_message_length', 1024 * 1024 * 12),
+                ('grpc.max_receive_message_length', 1024 * 1024 * 12),
+            ],
+        )
     else:
-        channel = grpc.insecure_channel(url)
+        channel = grpc.insecure_channel(url,
+            options=[
+                ('grpc.max_send_message_length', 1024 * 1024 * 12),
+                ('grpc.max_receive_message_length', 1024 * 1024 * 12),
+            ],
+        )
     try:
         grpc.channel_ready_future(channel).result(timeout=timeout)
         return channel
